Check for quoted-printable transfer encoding and

miaulalala opened the pull request:
decode messages before parsing html

Fixes https://github.com/nextcloud/mail/issues/7687

ilicz commented:
Tested the patch on my playground server and everything seems to be fine now.

ChristophWurst commented:
Could you check with the original approach in https://github.com/nextcloud/mail/commit/90a90dd24d21da7130cee7a514ca348f0de1fb24? Why did the decoding not work when we use Horde? I would prefer Horde decoding in the hopes that we address more encodings, if any, and not just quoted-printable.

ChristophWurst commented:
We also use the setTransferenCoding+getContents trick for attachments. That is why I wonder why it works only in some places.

miaulalala commented:
The issue comes from the `'decode' => true` on the fetch query. Removed that and added back the setContentTransferEncoding and setContent.
@ilicz I would apprechiate a second test if you will? :smile:

miaulalala commented:
uh looks like the setTransferEncoding is not possibe this way. I'll see how I can work this out.

ChristophWurst commented on lib/IMAP/MessageMapper.php:
```diff
@@ -726,11 +724,19 @@ public function getBodyStructureData(Horde_Imap_Client_Socket $client,
+				$mimeHeaders = $part->getMimeHeader($textBodyId, Horde_Imap_Client_Data_Fetch::HEADER_PARSE);
+				if ($enc = $mimeHeaders->getValue('content-transfer-encoding')) {
+					$structure->setTransferEncoding($enc);
+				}
```
I believe the ``$structure->setContents(...)`` and ``getContents()`` parts are missing

miaulalala commented:
Added the content check on the structure. it was indeed missing :sweat_smile:

ilicz commented:
I applied new patch and it seems the UTF-8 messages are fine now.
Sadly, I only now noticed meeting request sent from MS Outlook server - in iso8859-2. Such messages display correctly in reading pane, but the message preview in the message listing is not displayed correctly:

```
Content-Type: text/plain; charset="iso-8859-2"
Content-Transfer-Encoding: quoted-printable
```

![image](https://user-images.githubusercontent.com/98597308/205136269-b5982c2d-d63b-4906-9590-611e2829a14b.png)

kesselb commented:
I'm unsure if we are on the right track :thinking: 

The provided message works fine on my email server: https://github.com/nextcloud/mail/issues/7687#issuecomment-1335489245

ChristophWurst commented:
> The provided message works fine on my email server: [#7687 (comment)](https://github.com/nextcloud/mail/issues/7687#issuecomment-1335489245)

@miaulalala found out that if we request decoding when the message is fetched, the BINARY IMAP extension is required. And so depending on the capabilities of your server the result differs for the same message.

ChristophWurst commented on lib/IMAP/MessageMapper.php:
```diff
@@ -703,7 +703,6 @@ public function getBodyStructureData(Horde_Imap_Client_Socket $client,
 			$partsQuery = new Horde_Imap_Client_Fetch_Query();
 			if ($htmlBodyId !== null) {
 				$partsQuery->bodyPart($htmlBodyId, [
-					'decode' => true,
```
@miaulalala could you document your findings for later reference?

miaulalala replied:
Leaving a comment here so I don't forget. Not every IMIP server allows for the decdoing to happen there as it needs capability BINARY

kesselb commented:
- Added migration to reset preview text
- Rebased the pull request
- Documented our findings https://github.com/nextcloud/mail/issues/7746

ChristophWurst commented:
/backport to stable2.2

ChristophWurst commented:
/backport to stable1.14

ChristophWurst commented:
/backport to stable1.15
